static/js/message_list_data_cache.js

```diff
@@ -0,0 +1,84 @@
+import {Filter} from "./filter";
+import * as hash_util from "./hash_util";
+
+export class MessageListDataCache {
+    constructor() {
+        this._objects = new Map();
+    }
+
+    _get_key(filter) {
+        // The MLDCache API uses the filter object as the key.
+        // Internally, we serialize it, so this function
+        // should be called when only we want to access or
+        // create a stored MLD object.
+        return hash_util.operators_to_hash(filter.operators());
```

Comment:
I think this algorithm isn't deterministic, in that if you have the same operators in a different order, it won't work.

Reply:
(Or if I'm wrong and it is guaranteed to be sorted, we should have a comment explaining why, either here or in the `filter.operators()` definition)